### PR TITLE
[MNOE-1263] [OPAL-497] Updated local products provision button

### PR DESCRIPTION
**Pull request by RidaAhmad**
Updated local products provision button on the basis of currency

### CODE REVIEW

File: src/app/components/mno-local-products/mno-local-product.coffee [hunk 1]
```diff
@@ -23,6 +24,15 @@ angular.module 'mnoEnterpriseAngular'
     vm.hideNoPricingFound = (plan) ->
       vm.isPublic || vm.planAvailableForCurrency(plan) || !vm.pricedPlan(plan)
 
+    vm.buttonDisabled = () ->
```
Comment (iseessel) [inline src/app/components/mno-local-products/mno-local-product.coffee]: more specific naming -- perhaps `provisioningButtonDisabled`

File: src/app/components/mno-local-products/mno-local-product.coffee [hunk 2]
```diff
@@ -23,6 +24,15 @@ angular.module 'mnoEnterpriseAngular'
...
+      !vm.canProvision || !vm.orderPossible
+
+    vm.updateButtonDisabledTooltip = () ->
```
Comment (iseessel) [inline src/app/components/mno-local-products/mno-local-product.coffee]: naming

File: src/app/components/mno-local-products/mno-local-product.coffee (hunk 1 above)
Comment (iseessel) [reply src/app/components/mno-local-products/mno-local-product.coffee]: Might be clearer if we did: 

```
vm.provisioningButtonEnabled = () -> 
vm.canProvision && vm.orderPossible 
```

And update accordingly below.

File: src/app/components/mno-local-products/mno-local-product.coffee [hunk 3]
```diff
@@ -42,6 +52,13 @@ angular.module 'mnoEnterpriseAngular'
           productId = $stateParams.productId
           vm.product = _.findWhere(vm.products, { nid: productId })
           vm.product ||= _.findWhere(vm.products, { id: productId })
+          # Is currency selection enabled
+          currencySelection = MnoeConfig.isCurrencySelectionEnabled()
+          # Are there any available plans
+          availablePlans = ProvisioningHelper.plansForCurrency(vm.product.pricing_plans, vm.orgCurrency)
+
+          vm.orderPossible = !_.isEmpty(availablePlans) || (vm.product.pricing_plans&[0].default&[0] && currencySelection)
```
Comment (iseessel) [inline src/app/components/mno-local-products/mno-local-product.coffee]: save naviagtion operator in coffeescript is `?`

Comment (iseessel) [inline src/app/components/mno-local-products/mno-local-product.coffee]: Safe navigation in coffeescript is `?` 

vm.product.pricing_plans?[0].default?[0

**Comment by RidaAhmad** [comment]
Incorporated feedback regarding operator, thanks.
However, after discussion we've decided to keep the variable names same for now, though they don't really sound reasonable, but just to keep the local product & app install code consistent as same naming convention is used elsewhere, thank you.

**Pull request merged**